spec/system/user/user_updates_announcements_spec.rb

```diff
@@ -0,0 +1,22 @@
+require "rails_helper"
+
+RSpec.describe "Announcements", type: :system do
+  let(:user) { create(:user) }
+
+  before do
+    sign_in(user)
+    visit "/settings/misc"
+  end
+
+  it "renders the announcements settings" do
+    expect(page).to have_text("Announcements")
+  end
+
+  it "properly updates the announcements settings", js: true do
+    page.check "Display Announcements (When browsing)"
+    sidekiq_perform_enqueued_jobs do
+      click_button("Save Announcements Settings")
+    end
+    expect(page).to have_text("Your profile was successfully updated.")
```

Comment:
Hm, this test checks that the flash message shows up, but if you think about it, it's not really testing that the broadcast was updated. For example, someone could go into the broadcasts controller, make a change to the `save!` method, and this flash message might still show up (meaning the test would pass even though the code didn't actually work!).

I think you want to take a look at `spec/requests/user/user_settings_spec.rb`... 👀

Reply:
I think we eventually _do_ want to add a system spec for announcements in order to test if the announcement banner does not appear if you uncheck the "display announcements" box (but that feature isn't hooked up yet 😆)

Currently though, I don't think that currently these tests are really testing _your_ changes to the backend.